File: src/main/scala/ru/sbrf/gg/load/builder/HistSBookNoBuilder.scala

 package ru.sbrf.gg.load.builder

 import ru.sbrf.gg.load.TableInfo
 import ru.sbt.kmdtransform.TransformType._

 class HistSBookNoBuilder extends ObjectBuilder {
     def build(line: Array[String], tableInfo: TableInfo): Any = {
         val r = new com.sbt.cdm.model.extension.deposit.HistSBookNo()

         r.id = LONG.fromStr(line(0)).asInstanceOf[java.lang.Long]
r.objectId = LONG.fromStr(line(0)).asInstanceOf[java.lang.Long]
r.colocationKey = LONG.fromStr(line(1)).asInstanceOf[java.lang.Long]
r.rootParticle_DPL_id = LONG.fromStr(line(2)).asInstanceOf[java.lang.Long]
r.operationRun_DPL_id = LONG.fromStr(line(3)).asInstanceOf[java.lang.Long]
r.sbookBalance = BIGDECIMAL.fromStr(line(4)).asInstanceOf[java.math.BigDecimal]
r.sbook_branchno = LONG.fromStr(line(5)).asInstanceOf[java.lang.Long]
r.sbookCancelClerk = LONG.fromStr(line(6)).asInstanceOf[java.lang.Long]
r.sbookCancelDate = DATE_TIME.fromStr(line(7)).asInstanceOf[java.util.Date]
r.sbookCancelFirstnameClerk = STRING.fromStr(line(8)).asInstanceOf[java.lang.String]
r.sbookCancelSecondnameClerk = STRING.fromStr(line(9)).asInstanceOf[java.lang.String]
r.sbookCancelSurnameClerk = STRING.fromStr(line(10)).asInstanceOf[java.lang.String]
r.sbookDateClose = DATE_TIME.fromStr(line(11)).asInstanceOf[java.util.Date]
r.sbookDeclarationBranchno = LONG.fromStr(line(12)).asInstanceOf[java.lang.Long]
r.sbookDeclarationClerk = LONG.fromStr(line(13)).asInstanceOf[java.lang.Long]
r.sbookDeclarationDateLost = DATE_TIME.fromStr(line(14)).asInstanceOf[java.util.Date]
r.sbookDeclarationFirstname = STRING.fromStr(line(15)).asInstanceOf[java.lang.String]
r.sbookDeclarationKind = LONG.fromStr(line(16)).asInstanceOf[java.lang.Long]
r.sbookDeclarationNo = STRING.fromStr(line(17)).asInstanceOf[java.lang.String]
r.sbookDeclarationOffice = LONG.fromStr(line(18)).asInstanceOf[java.lang.Long]
r.sbookDeclarationSecondname = STRING.fromStr(line(19)).asInstanceOf[java.lang.String]
r.sbookDeclarationSurname = STRING.fromStr(line(20)).asInstanceOf[java.lang.String]
r.sbookDeclarationValid = LONG.fromStr(line(21)).asInstanceOf[java.lang.Long]
r.sbookDocumentkind = LONG.fromStr(line(22)).asInstanceOf[java.lang.Long]
r.sbookDubleClerk = LONG.fromStr(line(23)).asInstanceOf[java.lang.Long]
r.sbookDubleDate = DATE_TIME.fromStr(line(24)).asInstanceOf[java.util.Date]
r.sbookDubleFirstnameClerk = STRING.fromStr(line(25)).asInstanceOf[java.lang.String]
r.sbookDubleSecondnameClerk = STRING.fromStr(line(26)).asInstanceOf[java.lang.String]
r.sbookDubleSurnameClerk = STRING.fromStr(line(27)).asInstanceOf[java.lang.String]
r.sbookKind = LONG.fromStr(line(28)).asInstanceOf[java.lang.Long]
r.sbookNo = STRING.fromStr(line(29)).asInstanceOf[java.lang.String]
r.sbookPrintableno = STRING.fromStr(line(30)).asInstanceOf[java.lang.String]
r.sbookPrintablenoNew = STRING.fromStr(line(31)).asInstanceOf[java.lang.String]
r.sbookSbookno = STRING.fromStr(line(32)).asInstanceOf[java.lang.String]
r.sbookSbooknoDuble = STRING.fromStr(line(33)).asInstanceOf[java.lang.String]
r.sbookSeries = STRING.fromStr(line(34)).asInstanceOf[java.lang.String]
r.category_DPL_id = LONG.fromStr(line(35)).asInstanceOf[java.lang.Long]
r.partition_DPL_id = LONG.fromStr(line(36)).asInstanceOf[java.lang.Long]

         r
     }
 }
            

File: src/main/java/ru/sbrf/gg/load/builder/HistSBookNoBuilder.java

 package ru.sbrf.gg.load.builder;

 import ru.sbrf.gg.load.TableInfo;
 import com.sbt.DelimetedStringParser;
 import com.sbt.CompareUtils;
 import ru.sbrf.gg.load.builder.ObjectBuilder;
 import com.sbt.cdm.model.extension.deposit.HistSBookNo;

 import ru.sbt.kmdtransform.TransformType.*;

 public class HistSBookNoBuilder implements ObjectBuilder {
     @Override public Object build(String line, TableInfo tableInfo) {
         HistSBookNo r = new HistSBookNo();
         int[] indexes = new int[]{0, 0, line.length()};

         r.id = DelimetedStringParser._long(0, line, indexes);
r.objectId = DelimetedStringParser._long(0, line, indexes);
r.colocationKey = DelimetedStringParser._long(1, line, indexes);
r.rootParticle_DPL_id = DelimetedStringParser._long(2, line, indexes);
r.operationRun_DPL_id = DelimetedStringParser._long(3, line, indexes);
r.sbookBalance = DelimetedStringParser.bigDecimal(4, line, indexes);
r.sbook_branchno = DelimetedStringParser._long(5, line, indexes);
r.sbookCancelClerk = DelimetedStringParser._long(6, line, indexes);
r.sbookCancelDate = DelimetedStringParser.date(7, line, indexes);
r.sbookCancelFirstnameClerk = DelimetedStringParser.str(8, line, indexes);
r.sbookCancelSecondnameClerk = DelimetedStringParser.str(9, line, indexes);
r.sbookCancelSurnameClerk = DelimetedStringParser.str(10, line, indexes);
r.sbookDateClose = DelimetedStringParser.date(11, line, indexes);
r.sbookDeclarationBranchno = DelimetedStringParser._long(12, line, indexes);
r.sbookDeclarationClerk = DelimetedStringParser._long(13, line, indexes);
r.sbookDeclarationDateLost = DelimetedStringParser.date(14, line, indexes);
r.sbookDeclarationFirstname = DelimetedStringParser.str(15, line, indexes);
r.sbookDeclarationKind = DelimetedStringParser._long(16, line, indexes);
r.sbookDeclarationNo = DelimetedStringParser.str(17, line, indexes);
r.sbookDeclarationOffice = DelimetedStringParser._long(18, line, indexes);
r.sbookDeclarationSecondname = DelimetedStringParser.str(19, line, indexes);
r.sbookDeclarationSurname = DelimetedStringParser.str(20, line, indexes);
r.sbookDeclarationValid = DelimetedStringParser._long(21, line, indexes);
r.sbookDocumentkind = DelimetedStringParser._long(22, line, indexes);
r.sbookDubleClerk = DelimetedStringParser._long(23, line, indexes);
r.sbookDubleDate = DelimetedStringParser.date(24, line, indexes);
r.sbookDubleFirstnameClerk = DelimetedStringParser.str(25, line, indexes);
r.sbookDubleSecondnameClerk = DelimetedStringParser.str(26, line, indexes);
r.sbookDubleSurnameClerk = DelimetedStringParser.str(27, line, indexes);
r.sbookKind = DelimetedStringParser._long(28, line, indexes);
r.sbookNo = DelimetedStringParser.str(29, line, indexes);
r.sbookPrintableno = DelimetedStringParser.str(30, line, indexes);
r.sbookPrintablenoNew = DelimetedStringParser.str(31, line, indexes);
r.sbookSbookno = DelimetedStringParser.str(32, line, indexes);
r.sbookSbooknoDuble = DelimetedStringParser.str(33, line, indexes);
r.sbookSeries = DelimetedStringParser.str(34, line, indexes);
r.category_DPL_id = DelimetedStringParser._long(35, line, indexes);
r.partition_DPL_id = DelimetedStringParser._long(36, line, indexes);

         return r;
     }

     @Override public int compare(Object first, Object second) {
         HistSBookNo f = (HistSBookNo)first;
         HistSBookNo s = (HistSBookNo)second;

         if (CompareUtils.compare(f.id, s.id) != 0) return CompareUtils.compare(f.id, s.id);
if (CompareUtils.compare(f.objectId, s.objectId) != 0) return CompareUtils.compare(f.objectId, s.objectId);
if (CompareUtils.compare(f.colocationKey, s.colocationKey) != 0) return CompareUtils.compare(f.colocationKey, s.colocationKey);
if (CompareUtils.compare(f.rootParticle_DPL_id, s.rootParticle_DPL_id) != 0) return CompareUtils.compare(f.rootParticle_DPL_id, s.rootParticle_DPL_id);
if (CompareUtils.compare(f.operationRun_DPL_id, s.operationRun_DPL_id) != 0) return CompareUtils.compare(f.operationRun_DPL_id, s.operationRun_DPL_id);
if (CompareUtils.compare(f.sbookBalance, s.sbookBalance) != 0) return CompareUtils.compare(f.sbookBalance, s.sbookBalance);
if (CompareUtils.compare(f.sbook_branchno, s.sbook_branchno) != 0) return CompareUtils.compare(f.sbook_branchno, s.sbook_branchno);
if (CompareUtils.compare(f.sbookCancelClerk, s.sbookCancelClerk) != 0) return CompareUtils.compare(f.sbookCancelClerk, s.sbookCancelClerk);
if (CompareUtils.compare(f.sbookCancelDate, s.sbookCancelDate) != 0) return CompareUtils.compare(f.sbookCancelDate, s.sbookCancelDate);
if (CompareUtils.compare(f.sbookCancelFirstnameClerk, s.sbookCancelFirstnameClerk) != 0) return CompareUtils.compare(f.sbookCancelFirstnameClerk, s.sbookCancelFirstnameClerk);
if (CompareUtils.compare(f.sbookCancelSecondnameClerk, s.sbookCancelSecondnameClerk) != 0) return CompareUtils.compare(f.sbookCancelSecondnameClerk, s.sbookCancelSecondnameClerk);
if (CompareUtils.compare(f.sbookCancelSurnameClerk, s.sbookCancelSurnameClerk) != 0) return CompareUtils.compare(f.sbookCancelSurnameClerk, s.sbookCancelSurnameClerk);
if (CompareUtils.compare(f.sbookDateClose, s.sbookDateClose) != 0) return CompareUtils.compare(f.sbookDateClose, s.sbookDateClose);
if (CompareUtils.compare(f.sbookDeclarationBranchno, s.sbookDeclarationBranchno) != 0) return CompareUtils.compare(f.sbookDeclarationBranchno, s.sbookDeclarationBranchno);
if (CompareUtils.compare(f.sbookDeclarationClerk, s.sbookDeclarationClerk) != 0) return CompareUtils.compare(f.sbookDeclarationClerk, s.sbookDeclarationClerk);
if (CompareUtils.compare(f.sbookDeclarationDateLost, s.sbookDeclarationDateLost) != 0) return CompareUtils.compare(f.sbookDeclarationDateLost, s.sbookDeclarationDateLost);
if (CompareUtils.compare(f.sbookDeclarationFirstname, s.sbookDeclarationFirstname) != 0) return CompareUtils.compare(f.sbookDeclarationFirstname, s.sbookDeclarationFirstname);
if (CompareUtils.compare(f.sbookDeclarationKind, s.sbookDeclarationKind) != 0) return CompareUtils.compare(f.sbookDeclarationKind, s.sbookDeclarationKind);
if (CompareUtils.compare(f.sbookDeclarationNo, s.sbookDeclarationNo) != 0) return CompareUtils.compare(f.sbookDeclarationNo, s.sbookDeclarationNo);
if (CompareUtils.compare(f.sbookDeclarationOffice, s.sbookDeclarationOffice) != 0) return CompareUtils.compare(f.sbookDeclarationOffice, s.sbookDeclarationOffice);
if (CompareUtils.compare(f.sbookDeclarationSecondname, s.sbookDeclarationSecondname) != 0) return CompareUtils.compare(f.sbookDeclarationSecondname, s.sbookDeclarationSecondname);
if (CompareUtils.compare(f.sbookDeclarationSurname, s.sbookDeclarationSurname) != 0) return CompareUtils.compare(f.sbookDeclarationSurname, s.sbookDeclarationSurname);
if (CompareUtils.compare(f.sbookDeclarationValid, s.sbookDeclarationValid) != 0) return CompareUtils.compare(f.sbookDeclarationValid, s.sbookDeclarationValid);
if (CompareUtils.compare(f.sbookDocumentkind, s.sbookDocumentkind) != 0) return CompareUtils.compare(f.sbookDocumentkind, s.sbookDocumentkind);
if (CompareUtils.compare(f.sbookDubleClerk, s.sbookDubleClerk) != 0) return CompareUtils.compare(f.sbookDubleClerk, s.sbookDubleClerk);
if (CompareUtils.compare(f.sbookDubleDate, s.sbookDubleDate) != 0) return CompareUtils.compare(f.sbookDubleDate, s.sbookDubleDate);
if (CompareUtils.compare(f.sbookDubleFirstnameClerk, s.sbookDubleFirstnameClerk) != 0) return CompareUtils.compare(f.sbookDubleFirstnameClerk, s.sbookDubleFirstnameClerk);
if (CompareUtils.compare(f.sbookDubleSecondnameClerk, s.sbookDubleSecondnameClerk) != 0) return CompareUtils.compare(f.sbookDubleSecondnameClerk, s.sbookDubleSecondnameClerk);
if (CompareUtils.compare(f.sbookDubleSurnameClerk, s.sbookDubleSurnameClerk) != 0) return CompareUtils.compare(f.sbookDubleSurnameClerk, s.sbookDubleSurnameClerk);
if (CompareUtils.compare(f.sbookKind, s.sbookKind) != 0) return CompareUtils.compare(f.sbookKind, s.sbookKind);
if (CompareUtils.compare(f.sbookNo, s.sbookNo) != 0) return CompareUtils.compare(f.sbookNo, s.sbookNo);
if (CompareUtils.compare(f.sbookPrintableno, s.sbookPrintableno) != 0) return CompareUtils.compare(f.sbookPrintableno, s.sbookPrintableno);
if (CompareUtils.compare(f.sbookPrintablenoNew, s.sbookPrintablenoNew) != 0) return CompareUtils.compare(f.sbookPrintablenoNew, s.sbookPrintablenoNew);
if (CompareUtils.compare(f.sbookSbookno, s.sbookSbookno) != 0) return CompareUtils.compare(f.sbookSbookno, s.sbookSbookno);
if (CompareUtils.compare(f.sbookSbooknoDuble, s.sbookSbooknoDuble) != 0) return CompareUtils.compare(f.sbookSbooknoDuble, s.sbookSbooknoDuble);
if (CompareUtils.compare(f.sbookSeries, s.sbookSeries) != 0) return CompareUtils.compare(f.sbookSeries, s.sbookSeries);
if (CompareUtils.compare(f.category_DPL_id, s.category_DPL_id) != 0) return CompareUtils.compare(f.category_DPL_id, s.category_DPL_id);
if (CompareUtils.compare(f.partition_DPL_id, s.partition_DPL_id) != 0) return CompareUtils.compare(f.partition_DPL_id, s.partition_DPL_id);

         return 0;
     }
 }
            
